cfme/tests/cloud/test_quota.py

```diff
@@ -0,0 +1,241 @@
+                                     parent=new_tenant)
+    yield child_tenant
+    if child_tenant.exists:
+        child_tenant.delete()
+
+
+@pytest.fixture(scope='module')
+def new_group_child(appliance, new_child, new_tenant):
+    collection = appliance.collections.groups
+    group = collection.create(description='group_{}'.format(fauxfactory.gen_alphanumeric()),
+                              role='EvmRole-super_administrator',
+                              tenant='My Company/{}/{}'.format(new_tenant.name, new_child.name))
+    yield group
+    if group.exists:
+        group.delete()
+
+
+@pytest.fixture(scope='module')
+def new_user_child(appliance, new_group_child, new_credential):
+    collection = appliance.collections.users
+    user = collection.create(
+        name='user_{}'.format(fauxfactory.gen_alphanumeric()),
+        credential=new_credential,
+        email='xyz@redhat.com',
+        groups=new_group_child,
+        cost_center='Workload',
+        value_assign='Database')
+    yield user
+    if user.exists:
+        user.delete()
+
+
+@pytest.fixture(scope='module')
+def new_project(appliance):
+    collection = appliance.collections.projects
+    project = collection.create(name='project{}'.format(fauxfactory.gen_alphanumeric()),
+                                description='project_des{}'.format(fauxfactory.gen_alphanumeric()),
+                                parent=collection.get_root_tenant())
+    yield project
+    if project.exists:
+        project.delete()
+
+
+@pytest.fixture(scope='module')
+def new_group_project(appliance, new_project):
+    collection = appliance.collections.groups
+    group = collection.create(description='group_{}'.format(fauxfactory.gen_alphanumeric()),
+                              role='EvmRole-super_administrator',
+                              tenant='My Company/{}'.format(new_project.name))
+    yield group
+    if group.exists:
+        group.delete()
+
+
+@pytest.fixture(scope='module')
+def new_user_project(appliance, new_group_project, new_credential):
+    collection = appliance.collections.users
+    user = collection.create(
+        name='user_{}'.format(fauxfactory.gen_alphanumeric()),
+        credential=new_credential,
+        email='xyz@redhat.com',
+        groups=new_group_project,
+        cost_center='Workload',
+        value_assign='Database')
+    yield user
+    if user.exists:
+        user.delete()
+
+
+# first arg of parametrize is the list of fixtures or parameters,
+# second arg is a list of lists, with each one a test is to be generated
+# sequence is important here
+# indirect is the list where we define which fixtures are to be passed values indirectly.
+# OpenStackProvider has infra issue, so skipping the test for it.
+@pytest.mark.uncollectif(lambda provider, appliance: provider.one_of(OpenStackProvider))
+@pytest.mark.parametrize(
+    ['set_child_tenant_quota', 'custom_prov_data', 'extra_msg', 'approve'],
```

Comment:
Like we talked about before can you add a comment that helps explain what 'custom_prov_data', 'extra_msg', and 'approve' are being used for in the test? And can the comment also explain that the tuple for `set_child_tenant_quota` will become the `request.param` in the `set_child_tenant_quota` fixture above.